src/HotChocolate/ApolloFederation/src/ApolloFederation/Helpers/EntitiesResolver.cs

```diff
@@ -16,34 +16,75 @@ internal static class EntitiesResolver
         IReadOnlyList<Representation> representations,
         IResolverContext context)
     {
-        var entities = new List<object?>();
+        Task<object?>[] tasks = ArrayPool<Task<object?>>.Shared.Rent(representations.Count);
+        tasks.AsSpan().Slice(0, representations.Count).Clear();
```

Comment:
@michaelstaib  i dont think this clearing is necessary as we anyway re-assign each index of `repesentations.Count` and only iterate until `representations.Count`

Reply:
Got it makes sense.

Reply:
We should clear at the end so that the GC can collect the tasks. We often also clear at the beginning if there is a possibility that we could access something old :)

If that is guaranteed to not happen because we are setting everything then fine...

Reply:
Also make sure to use the indexer only once per iteration.

Reply:
@michaelstaib  accessing the indexer once per iteration is not really possible. Coudl only reduce it by one access for the first loop.
Array is now cleared by. calling .Return(tasks, true) and not cleared in the beginning.